### PR TITLE
upgrades dev dependencies

### PR DESCRIPTION
**Pull request by chrabyrd**

### CODE REVIEW

**Comment by chrabyrd** [comment]
The one update that requires intervention is to the `eslint` package. `.eslintrc.js` and `.eslintignore` must be deleted and `eslint.config.mjs` must be ported over from `arches/install/arches-templates`

File: package.json [hunk 1]
```diff
@@ -2,51 +2,56 @@
...
+        "vue-tsc": "^2.0.13",
+        "webpack": "^5.91.0",
+        "webpack-cli": "^5.1.4",
+        "webpack-dev-server": "^5.0.4",
+        "webpack-merge": "^5.10.0"
+    },
+    "resolutions": {
+        "eslint": "^8.56.0",
+        "vite": "5.0.0",
+        "vue": "^3.2.25"
```
Comment (jacobtylerwalls) [inline package.json]: Thanks. I've only just learned what peer dependencies are. I found [this link](https://www.johno.com/using-peer-dependencies) helpful.

Still being on v1 yarn (lol) it's possible this is too noisy / incorrect, but with the latest, I'm getting:
```js
warning "arches-dev-dependencies > @babel/eslint-parser@7.24.1" has incorrect peer dependency "eslint@^7.5.0 || ^8.0.0".
warning "arches-dev-dependencies > @typescript-eslint/parser@7.7.0" has incorrect peer dependency "eslint@^8.56.0".
warning "arches-dev-dependencies > @vitejs/plugin-vue@5.0.4" has unmet peer dependency "vite@^5.0.0".
warning "arches-dev-dependencies > typescript-eslint@7.7.0" has incorrect peer dependency "eslint@^8.56.0".
warning "arches-dev-dependencies > typescript-eslint > @typescript-eslint/eslint-plugin@7.7.0" has incorrect peer dependency "eslint@^8.56.0".
warning "arches-dev-dependencies > typescript-eslint > @typescript-eslint/utils@7.7.0" has incorrect peer dependency "eslint@^8.56.0".
warning "arches-dev-dependencies > typescript-eslint > @typescript-eslint/eslint-plugin > @typescript-eslint/type-utils@7.7.0" has incorrect peer dependency "eslint@^8.56.0".
```

We can probably ignore the "incorrect" warnings as noise, but the one "unmet" warning says we need to resolve to vite 5.0.0.

```
$ yarn list vite
yarn list v1.22.19
warning Filtering by arguments is deprecated. Please use the pattern option instead.
└─ vite@5.2.9
✨  Done in 0.20s.
```

Thoughts? Possible?

Comment (chrabyrd) [reply package.json]: Yeah those resolution statement don't appear to do anything 😞 

IMO this is all just noise that will hopefully go away when we update/change package managers. I'm unsure if there's any actions we could/should do to resolve this. What do you think?

Comment (jacobtylerwalls) [reply package.json]: Looks like it's a busy time for tooling with the eslint 9 release.

- `@babel/eslint-parser` somewhere between 7.24.1 > 7.24.5 adds eslint 9 as acceptable peer, so let's take that
- `@typescript-eslint`: we [need](https://github.com/typescript-eslint/typescript-eslint/issues/8211) to be on v8-alpha-10 or newer
- `@vitejs/plugin-vue@5.0.4` wants us to take vue as a dep, which makes sense

See diff in next comment that worked for me

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
I'm showing the warnings are fixed with the following, save for the one bug in `@typescript-eslint/eslint-plugin` that was [just patched](https://github.com/typescript-eslint/typescript-eslint/pull/9089):

```diff
diff --git a/package.json b/package.json
index 613f46f..68cc597 100644
--- a/package.json
+++ b/package.json
@@ -3,12 +3,12 @@
     "version": "7.6.0",
     "dependencies": {
         "@babel/core": "^7.24.4",
-        "@babel/eslint-parser": "^7.24.1",
+        "@babel/eslint-parser": "^7.24.5",
         "@babel/plugin-syntax-dynamic-import": "^7.8.3",
         "@babel/plugin-transform-class-properties": "^7.24.1",
         "@babel/plugin-transform-runtime": "^7.24.3",
         "@babel/preset-env": "^7.24.2",
-        "@typescript-eslint/parser": "^7.7.0",
+        "@typescript-eslint/parser": "^8.0.0-alpha.10",
         "@vitejs/plugin-vue": "^5.0.4",
         "@vitest/coverage-v8": "^1.5.0",
         "@vue/test-utils": "^2.4.5",
@@ -39,8 +39,10 @@
         "text-loader": "^0.0.1",
         "ts-loader": "^9.5.1",
         "typescript": "^5.4.5",
-        "typescript-eslint": "^7.7.0",
+        "typescript-eslint": "^8.0.0-alpha.10",
+        "vite": "^5.0.0",
         "vitest": "^1.5.0",
+        "vue": "^3.4.21",
         "vue-loader": "^17.4.2",
         "vue-template-compiler": "^2.7.16",
         "vue-tsc": "^2.0.13",
```

**Review by jacobtylerwalls — APPROVED** [review]

**Pull request merged**